helm-projectile.el

```diff
@@ -368,6 +368,69 @@ CANDIDATE is the selected file.  Used when no file is explicitly marked."
                                             (list candidate))))
           (rename-buffer dired-buffer-name))))))
 
+
+(defun helm-projectile-find-file-or-marked (candidate)
```

Comment:
That would probably work. I haven't used advices in elisp, and I'm afraid my knowledge of them is limited.